### PR TITLE
fixed calendso busy query timezone issue

### LINKED ISSUE #467: CalDAV timezone seems to be incorrect (Nextcloud/Zimbra CalDAV server)
Found a bug? Please fill out the sections below. 👍

### Issue Summary
A summary of the issue. This needs to be a clear detailed-rich summary. 

### Steps to Reproduce
1. I have set all my relevant timezones to Amsterdam
2. Connect new CalDAV Calendar
3. I see my availability, but the hours are wrong, the appointment is at 14:00, but is at 16:00 in calendso

Screenshot from 2021-08-17 10-58-58
Screenshot from 2021-08-17 10-59-23
Screenshot from 2021-08-17 10-59-59

### PR DESCRIPTION
**Pull request by natelindev**
## What does this PR do?

fixed calendso busy query timezone issue

Fixes #467 (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code and corrected any misspellings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/BaseCalendarApiAdapter.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Credential } from "@prisma/client";
 import dayjs from "dayjs";
+import tz from "dayjs/plugin/timezone";
```
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:3]: Call this `timezone` for consistency

File: lib/BaseCalendarApiAdapter.ts [hunk 2]
```diff
@@ -304,10 +306,10 @@ export class BaseCalendarApiAdapter {
             vcalendar.getFirstSubcomponent("vtimezone")?.getFirstPropertyValue("tzid") || "";
 
           const startDate = calendarTimezone
-            ? dayjs(event.startDate.toJSDate()).tz(calendarTimezone)
+            ? dayjs.tz(event.startDate.toString(), calendarTimezone)
```
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:309]: I don't think you need toString() here

**Review by emrysal — COMMENTED** [review]
Couple of amends but it definitely did not work like it was, good spot.

File: lib/BaseCalendarApiAdapter.ts (hunk 2 above)
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:309]:
```suggestion
            ? dayjs.tz(event.startDate, calendarTimezone)
```

File: lib/BaseCalendarApiAdapter.ts [hunk 3]
```diff
@@ -304,10 +306,10 @@ export class BaseCalendarApiAdapter {
...
             : new Date(event.startDate.toUnixTime() * 1000);
           const endDate = calendarTimezone
-            ? dayjs(event.endDate.toJSDate()).tz(calendarTimezone)
+            ? dayjs.tz(event.endDate.toString(), calendarTimezone)
```
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:312]:
```suggestion
            ? dayjs.tz(event.endDate, calendarTimezone)
```

File: lib/BaseCalendarApiAdapter.ts (hunk 1 above)
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:3]:
```suggestion
import timezone from "dayjs/plugin/timezone";
```

File: lib/BaseCalendarApiAdapter.ts [hunk 4]
```diff
@@ -21,6 +22,7 @@ import logger from "@lib/logger";
 import { CalendarEvent, IntegrationCalendar } from "./calendarClient";
 
 dayjs.extend(utc);
+dayjs.extend(tz);
```
Comment (emrysal) [inline lib/BaseCalendarApiAdapter.ts:25]:
```suggestion
dayjs.extend(timezone);
```

File: lib/BaseCalendarApiAdapter.ts (hunk 2 above)
Comment (natelindev) [reply lib/BaseCalendarApiAdapter.ts:309]: without toString ts will complain
Screen Shot 2021-12-27 at 8 19 45 PM

**Comment by natelindev** [comment]
Any re-reviews?

**Review by emrysal — APPROVED** [review]

**Comment by emrysal** [comment]
LGTM, good pr

**Pull request merged**